go/worker/storage/p2p/checkpointsync/protocol.go

```diff
@@ -65,7 +65,7 @@ type GetCheckpointChunkResponse struct {
 func init() {
 	peermgmt.RegisterNodeHandler(&peermgmt.NodeHandlerBundle{
 		ProtocolsFn: func(n *node.Node, chainContext string) []core.ProtocolID {
-			if !n.HasRoles(node.RoleComputeWorker | node.RoleStorageRPC) {
+			if !n.HasRoles(node.RoleComputeWorker | node.RoleStorageRPC | node.RoleObserver) {
```

Comment:
```suggestion
			if !n.HasRoles(node.RoleComputeWorker | node.RoleObserver | node.RoleStorageRPC) {
```
If possible, I try to keep the same order as in enum, so that it is easier to check which roles are missing. Huge nit :D